lib/libutil++/libutil++

```diff
@@ -0,0 +1,240 @@
+/*-
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2025 Chelsio Communications, Inc.
+ * Written by: John Baldwin <jhb@FreeBSD.org>
+ */
+
+#ifndef __LIBUTILPP__
+#define	__LIBUTILPP__
+
+#include <sys/cdefs.h>
+#include <sys/nv.h>
+#include <libutil.h>
+#include <netdb.h>
+#include <stdarg.h>
+#include <unistd.h>
+
+#include <memory>
+
+namespace freebsd {
+	/*
+	 * FILE_up is a std::unique_ptr<> for FILE objects which uses
+	 * fclose() to destroy the wrapped pointer.
+	 */
+	struct fclose_deleter {
+		void operator() (FILE *fp) const
+		{
+			fclose(fp);
+		}
+	};
+
+	typedef std::unique_ptr<FILE, fclose_deleter> FILE_up;
+
+	/*
+	 * addrinfo_up is a std::unique_ptr<> which uses
+	 * freeaddrinfo() to destroy the wrapped pointer.  It is
+	 * intended to wrap arrays allocated by getaddrinfo().
+	 */
+	struct freeaddrinfo_deleter {
+		void operator() (struct addrinfo *ai) const
+		{
+			freeaddrinfo(ai);
+		}
+	};
+
+	typedef std::unique_ptr<addrinfo, freeaddrinfo_deleter> addrinfo_up;
+
+	/*
+	 * This class is intended to function similar to unique_ptr<>,
+	 * but it contains a file descriptor rather than a pointer to
+	 * an object.  On destruction the descriptor is closed via
+	 * close(2).
+	 *
+	 * Similar to unique_ptr<>, release() returns ownership of the
+	 * file descriptor to the caller.  reset() closes the current
+	 * file descriptor and takes ownership of a new one.  A move
+	 * constructor permits ownership to be transferred via
+	 * std::move().  An integer file descriptor can be assigned
+	 * directly which is equivalent to calling reset().
+	 *
+	 * An explicit bool conversion operator permits testing this
+	 * class in logical expressions.  It returns true if it
+	 * contains a valid descriptor.
+	 *
+	 * An implicit int conversion operator returns the underlying
+	 * file descriptor allowing objects of this type to be passed
+	 * directly to APIs such as connect(), listen(), etc.
+	 */
+	class fd_up {
+	public:
+		fd_up() : _fd(-1) {}
+		fd_up(int fd) : _fd(fd) {}
+		fd_up(fd_up &&other) : _fd(other.release()) {}
+		fd_up(fd_up const &) = delete;
+
+		~fd_up() { reset(); }
+
+		int
+		get() const
+		{
+			return (_fd);
```

Comment:
I think this is too verbose. You can do it one one line. And w/o the () in the return

Reply:
i.e. `int get() const { return _fd; }`?